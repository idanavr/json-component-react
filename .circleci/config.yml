version: 2.1
jobs:
  test:
    docker:
      - image: circleci/node:13.6.0

    working_directory: ~/repo

    steps:
      - checkout
      - restore_cache:
          keys:
          - v1-dependencies-{{ checksum "package.json" }}
          # fallback to using the latest cache if no exact match is found
          - v1-dependencies-
      - run: npm install
      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}       
      - run: npm test

  publish:
    docker:
      - image: circleci/node:13.6.0
      
    working_directory: ~/repo

    steps:
      - checkout
      - restore_cache:
          keys:
          - v1-dependencies-{{ checksum "package.json" }}
          - v1-dependencies-
      - run: npm run build
      - run:
          name: Publish to NPM
          command: | 
            npm set //registry.npmjs.org/:_authToken=$NPM_TOKEN
            npm publish

workflows:
  version: 2.1
  test-publish:
    jobs:
      - test:
          filters:
            tags:
              only: /v[0-9]+(\.[0-9]+)*/
      - publish:
          requires:
            - test
          filters:
            branches:
              only:
                - master
            tags:
              only: /v[0-9]+(\.[0-9]+)*/            
            