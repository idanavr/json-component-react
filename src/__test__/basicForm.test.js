import React from 'react';
import renderer from 'react-test-renderer'; // ES6
import '@testing-library/jest-dom/extend-expect'; // ES6
import FormGenerator from '..';
import config from './testConfig';

it('matches snapshot', () => {
    const tree = renderer.create(<FormGenerator config={config} />).toJSON();
    expect(tree).toMatchSnapshot();
});