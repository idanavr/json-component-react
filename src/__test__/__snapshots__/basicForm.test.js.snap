// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="dynamic-form"
>
  <div
    className="dynamic-form-fields"
  >
    <div
      className="dynamic-field-component form-group"
    >
      <label
        className=""
      >
        Email
      </label>
      <input
        aria-invalid={false}
        className="form-control"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      
    </div>
    <div
      className="dynamic-field-component form-group"
    >
      <label
        className=""
        data-tip="First Name and Last Name"
      >
        Full Name
      </label>
      <input
        aria-invalid={false}
        className="form-control"
        onChange={[Function]}
        placeholder="Full Name"
        type="text"
        value=""
      />
      
    </div>
    <myInput
      className="my-input"
      onChange={[Function]}
      onClick={[Function]}
      placeholder="Habits placeholder"
    />
  </div>
  <div
    className="dynamic-form-controllers"
  >
    <button
      aria-label={null}
      className="dynamic-controller-component  btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
    <myBtn
      className="my-btn"
      onChange={[Function]}
      onClick={[Function]}
    >
      Clear
    </myBtn>
  </div>
</div>
`;
